tools/commands-packages-query.js

```diff
@@ -0,0 +1,1159 @@
+  //    - defaultVersion: version record for the default version of this release.
+  //    - totalVersions: total number of release versions for this track
+  //    - versions: an array of version objects. If only recommended versions
+  //      are returned, ordered by orderKey, otherwise unordered. Objects have
+  //      the following keys:
+  //         - version: version number
+  //         - description: version description
+  //         - recommended: true for recommended versions
+  //         - orderKey: (only if showHiddenVersions is true) the orderKey of
+  //           this version.
+  //         - publishedBy: username of the publisher
+  //         - publishedOn: date the version was published
+  _displayRelease: function (data) {
+    var self = this;
+
+    Console.info("Release:",  data.track);
+    // There is no such thing as a local release, which means all releases have
+    // a maintainer.
+    Console.info("Maintainers:", data.maintainers.join(", "));
+    Console.info();
+
+    if (data.defaultVersion) {
+      Console.info(data.defaultVersion.description);
+      Console.info();
+    }
+
+    if (self.showHiddenVersions) {
+      self._displayAllReleaseVersions(data.versions);
+      return;
+    }
+
+    // Display the recommended versions of this release.
+    var versionsShown = 0;
+    if (!_.isEmpty(data.versions)) {
+      Console.info("Recommended versions:");
+      var rows = [];
+      _.each(data.versions, function (v) {
+        rows.push([v.version, utils.longformDate(v.publishedOn)]);
+        versionsShown++;
+      });
+      Console.printTwoColumns(rows, { indent: 2 });
+    };
+
+    // Display a warning about other release versions at the bottom.
+    if (data.totalVersions > versionsShown) {
+      var versionsPluralizer =
+            (data.totalVersions > 1) ?
+            "all " + data.totalVersions + " versions" :
+            "the hidden version";
+      Console.info(
+        "Some versions of", self.name, "have been hidden.",
+        "To see " + versionsPluralizer + ", run",
+        Console.command("'meteor show --show-all " + self.name + "'") + ".");
+    }
+  },
+  // Displays all the versions of a given release in a human-readable
+  // format. Includes experimental and otherwise hidden versions. Takes in an
+  // array of version objects, each of which has the following keys:
+  //  - version: version string
+  //  - orderKey: (optional) orderKey of this version. Not all versions have
+  //    orderKeys.
+  //  - publishedOn: date of publication
+  //  - recommended: true if the version is recommended.
+  _displayAllReleaseVersions: function (versions) {
+    var self = this;
+    var columnOpts = { indent: 2, ignoreWidth: true };
+    // If we don't have any versions, then there is nothing to display.
+    if (! versions) { return; }
+
+    // We are going to print versions with order key ('versions'), separately
+    // from versions without an order key ('experimental versions').
```

Comment:
I think, in this instance, it kind of does? We spent a while thinking about this, which is not to say that we are right. But roughly:
-  we want to order versions with the order key by key! That makes sense. That includes pre-releases (1.0-rc.0) and deprecated versions (1.0). 
-  we also want to display versions without an order key, but we can't order them by key, because they don't have one. A conundrum.
Then, we thought about the circumstances when versions don't have order keys, and decided that it only really covers whacky one-of versions. We could call them 'experimental' versions. That doesn't mean that none of the versions under the normal Versions heading are, strictly-speaking, non-experimental, just that these versions are definitely weird.